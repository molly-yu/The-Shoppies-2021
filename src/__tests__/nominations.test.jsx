import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

import Nominations from '../components/nominations';
import Button from '../components/button';
import Icon from '../components/icon';

const movie1 = { imdbID: "1", Title: "title1", Year: "2000" };

describe('Nominations', () => {
  it("render nominations without crashing", () => {
    const props = {
        nominations: [movie1], 
        remove: jest.fn(),
        open: jest.fn()
    }
    const wrapper = shallow(<Nominations {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it("should remove nominee", () => {
    const remove = jest.fn();
    const props = {
        nominations: [movie1], 
        remove: remove,
        open: jest.fn()
    }
    const wrapper = shallow(<Nominations {...props} />);
    const button = wrapper.findWhere(b => b.is(Button) && b.prop('round') === true).first();
    button.simulate('click', {});
    expect(remove).toBeCalled();
  })

  it("should open modal", () => {
    const open = jest.fn();
    const props = {
        nominations: [movie1], 
        remove: jest.fn(),
        open: open
    }
    const wrapper = shallow(<Nominations {...props} />);
    const icon = wrapper.find(Icon).first();
    icon.simulate('click', {});
    expect(open).toBeCalled();
  })

 })