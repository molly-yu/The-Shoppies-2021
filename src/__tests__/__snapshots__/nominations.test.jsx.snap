// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nominations render nominations without crashing 1`] = `
<Fragment>
  <tr
    key="1"
  >
    <td>
      <styled.button
        onClick={[Function]}
        round={true}
      >
        ✕
      </styled.button>
    </td>
    <td>
      title1 (2000)
    </td>
    <td>
      <styled.button
        close={true}
      >
        <styled.img
          alt="Open"
          aria-label="Expand"
          onClick={[Function]}
          src="open.png"
        />
      </styled.button>
    </td>
  </tr>
</Fragment>
`;

exports[`SearchResults render searchresults without crashing 1`] = `
<Fragment>
  <tr
    key="1"
  >
    <td>
      <styled.button
        onClick={[Function]}
        round={true}
      >
        ✕
      </styled.button>
    </td>
    <td>
      title1 (2000)
    </td>
    <td>
      <styled.button
        close={true}
      >
        <styled.img
          alt="Open"
          aria-label="Expand"
          onClick={[Function]}
          src="open.png"
        />
      </styled.button>
    </td>
  </tr>
</Fragment>
`;
